### PR TITLE
Adding timeout for httpclient

### PR DESCRIPTION
**Pull request by rishabhbits038**
## Problem
Socket timeout was passed but connection timeouts were not kicking.

## Solution
Added timeouts in the httpclient

##### Does this solution apply anywhere else?
- [ ] yes
- [x] no

##### Testing done:
- [x] Unit tests
- [ ] Integration tests
- [ ] System tests
- [ ] Manual tests

### CODE REVIEW

File: src/main/java/com/splunk/hecclient/HttpClientBuilder.java [hunk 1]
```diff
@@ -74,6 +74,9 @@ public CloseableHttpClient build() {
                 .setSoTimeout(socketTimeout * 1000)
                 .build();
         RequestConfig requestConfig = RequestConfig.custom()
+                .setSocketTimeout(socketTimeout)
+                .setConnectionRequestTimeout(socketTimeout)
+                .setConnectTimeout(socketTimeout)
```
Comment (aniketshrimal) [inline src/main/java/com/splunk/hecclient/HttpClientBuilder.java:79]: I checked and `splunk.hec.socket.timeout` is in sec - thus we probably need to multiply by 1000 to ensure that units are correct. Could you please double check if it is required for each of these parameters settings. 
```suggestion
                .setSocketTimeout(socketTimeout*1000)
                .setConnectionRequestTimeout(socketTimeout*1000)
                .setConnectTimeout(socketTimeout*1000)
```

Comment (aniketshrimal) [reply src/main/java/com/splunk/hecclient/HttpClientBuilder.java:79]: Also another question is - do we want to have different configs or let sockettimeout control other parameters as well (like ConnectionRequestTimeout and ConnectTimeout)?

Comment (rishabhbits038) [reply src/main/java/com/splunk/hecclient/HttpClientBuilder.java]: Added connectionRequestTimeout and connectionTimeout as params

**Review by aniketshrimal — APPROVED** [review]
LGTM @rishabhbits038! Thanks for addressing the comments.

**Pull request merged**